src/helpers.js

```diff
@@ -234,7 +234,7 @@ function populateChild(state, child, p) {
   // populate child is key
   if (typeof childVal === 'string' || childVal instanceof String) {
     // attach child paramter if it exists
-    const dotRoot = p.root
+    const dotRoot = (p.storeAs ? p.storeAs : p.root)
```

Comment:
Nit pick: This could be `p.storeAs || p.root` right? Not a big deal though, I'll still get it merged

Reply:
Yes, this could've been much simpler